### ISSUE TITLE
Deletion of notices, translations?

### ISSUE THREAD

**Issue by cblgh** [ssb-ngi-pointer/go-ssb-room#83]
I was playing around on the development server I have locally, and noticed I accidentally created two translations with the same name (hm). Thing is, I can't delete the duplicate!

Have you @staltz / @cryptix talked about being able to remove notices as an admin?

**Comment by cryptix** [comment]
Hm... just for understanding is the title, language or pinned name duplicated?

I guess there should be a better constrained on the db to protect against duplicates.

For background: Initially I wanted to include the language in the relation which links them but somehow couldn't get it working as intended and then just punted on it with a classical n:m which uses just the IDs.

**Comment by cblgh** [comment]
the language is duplicated. pushed a fix in `notices.go` that prevents empty content/title in addition to id/language when adding translations: https://github.com/ssb-ngi-pointer/go-ssb-room/pull/64/commits/864f3a249ef74a5f020f6605d3b04ed00b0d531e should make accidentally creating a bogus notice and having it be saved less likely

**Comment by cblgh** [comment]
![2021-03-15-155739_1142x663_scrot](https://user-images.githubusercontent.com/3862362/111173900-49da2d80-85a7-11eb-9f5b-82daa99639c8.png)

**Comment by cryptix** [comment]
I was looking a bit at sqlite's docs and don't think it's possible to add a constraint that would protect.

The most simplest thing we should do then is adding a check to see if the lanauge is already there before the noticed is added to the _pinned notice_ here:

https://github.com/ssb-ngi-pointer/go-ssb-room/blob/c9eb1aef9dc0f867a71d2469c638342ed5d92fda/roomdb/sqlite/notices.go#L107

For better consistency, the hold thin should be wrapped into a `transact(...)`. Here is an example from the aliases where it checks that the feed is on the allow list (i.e. a member) before it registers it:

https://github.com/ssb-ngi-pointer/go-ssb-room/blob/ae816736f6c2dc61734b5ce5317f15e811379a9a/roomdb/sqlite/aliases.go#L81.

This should protect against having duplicates in the first place.

**Comment by ahdinosaur** [comment]
i support being able to delete notices. by default there's a "description" and "privacy policy" translated to "de-DE", i don't speak german so now the room is stuck with a bad translation. that being said, this is awesome! :smiley_cat:

**Comment by cryptix** [comment]
Yes, totally agree. I think our test cases leaked into the wild/first release.

I'd wager to collect #213, #161 and this issue to make this feature more coherent and useful.
